store/prefixstore.go

```diff
@@ -99,24 +99,16 @@ func (s prefixStore) Iterator(start, end []byte) Iterator {
 // Implements KVStore
 // Check https://github.com/tendermint/tendermint/blob/master/libs/db/prefix_db.go#L129
 func (s prefixStore) ReverseIterator(start, end []byte) Iterator {
-	var newstart []byte
-	if start == nil {
-		newstart = cpIncr(s.prefix)
-	} else {
-		newstart = cloneAppend(s.prefix, start)
-	}
+	newstart := cloneAppend(s.prefix, start)
 
 	var newend []byte
 	if end == nil {
-		newend = cpDecr(s.prefix)
+		newend = cpIncr(s.prefix)
```

Comment:
Hmmm, was this also a bug before?

Reply:
I think the upstream parameter semantics changed.